tox.ini

```diff
@@ -80,3 +80,17 @@ setenv =
     FLIT_USERNAME = __token__
 commands = flit publish
 skip_install = True
+
+# Ensure compatibility is maintained with protobuf 3.X
+[testenv:proto3]
+description = run tests with pytest with coverage
+deps =
+    pytest>=6.2.5,<7.0
+    pytest-cov>=2.10.1,<3.0
+    pytest-html>=3.1.1,<4.0
+    tls_test_tools>=0.1.1
+    wheel>=0.38.4
+commands =
+    pip uninstall grpcio-health-checking -y
+    pip install protobuf==3.19.0 grpcio-health-checking --upgrade
```

Comment:
🤔 does this fail with the simple: 
```
deps =
    protobuf==3.19.0
```?
And if so does that mean users can't easily install with protobuf 3.19 anyway?

Reply:
Ug, this is just me punting on figuring out how to do `tox` right. The issue here is that the installation of the project requirements happens _after_ the installation of things in the `deps` list, so what happens is:

* `deps` install
* Generic install against `pyproject.proto` happen
    * This installs the _latest_ of `protobuf` and `grpcio-health-checking`
    * The `latest` `grpcio-health-checking` requires protobuf 4.X, but earlier versions that are in our range allow 3.X

Because of this ordering, the two other things I tried don't work:

* Put `protobuf==3.19.0` in `deps`: This fails because it gets overwritten when the install of `pyrpoject.proto` happens
* Just `pip install protobuf==3.19.0 --upgrade` in the `commands`: This fails because `grpcio-health-checking` is already installed at a version that is incompatible with `3.X`